subsys/net/lib/nrf_cloud/src/nrf_cloud_fota.c

```diff
@@ -1492,3 +1492,20 @@ int nrf_cloud_fota_ble_job_update(const struct nrf_cloud_fota_ble_job
 	return publish_and_free_array(array, &topic_ble_updt, &param);
 }
 #endif
+
+#if defined(CONFIG_NRF_CLOUD_FOTA_MODEM_FW_UPDATE_AUTO_VALIDATE)
+static int init(const struct device *unused)
```

Comment:
for modem delta FOTA, the modem lib needs to be initialized first so the ret val can be checked.
perhaps the auto validate option should depend on `CONFIG_NRF_MODEM_LIB_SYS_INIT=y`?
if so, that would prevent it from being used with `CONFIG_NRF_CLOUD_FOTA_FULL_MODEM_UPDATE=y` since `FMFU_DEV` requires `!NRF_MODEM_LIB_SYS_INIT`  🙃 

otherwise, for fmfu, the flash device needs to be set first, as done here: 
https://github.com/nrfconnect/sdk-nrf/blob/6c8a0963b83d81fcedfaee777d73d02821e8ea61/applications/asset_tracker_v2/src/cloud/nrf_cloud_integration.c#L60

Reply:
Agreed!

Reply:
Ill add the Kconfig dependency and update the docs a bit before I take it out of draft.